```{r setup}
library(tidyverse)
library(DESeq2)
library(ChIPseeker)
library(clusterProfiler)
library(ComplexHeatmap)
library(colorRamp2)


id_symbol <- read_delim('/media/alessio/Data/genome110/ID_SYMBOL_converter.csv') |> 
  dplyr::select(gene_id_version, symbol)
```

#######################
# Differential analysis
#######################

### Restart from the salmon counts, we use the new samples only normoxia
```{r}
library(tximeta)

Experiment_salmon <- read_csv("Salmon_New_samples_RNAseq_samplesheet.csv") %>% 
  dplyr::mutate(files =  paste0("../RNAseq/salmon_quant/Samples/",files))

# only normoxia samples
Experiment_salmon <- Experiment_salmon %>% 
  dplyr::filter(No_myco == T & str_detect(Condition_time, "_0"))

se <- tximeta(Experiment_salmon)
se <- summarizeToGene(se)

DDS <- DESeqDataSet(se, design = ~ Condition_time)

DDS <- DESeq(DDS)

VST <- vst(DDS, blind = F)


plotPCA(VST, intgroup = "Condition_time")

H4KO_Vs_Cas9_norm_Up <- results(DDS, contrast = c("Condition_time", "H4KO_0", "Cas9_0")) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id_version") %>% 
  left_join(id_symbol, by = "gene_id_version") %>% 
  dplyr::filter(log2FoldChange > 0.75 & padj < 0.05)

H4KO_Vs_Cas9_norm_Down <- results(DDS, contrast = c("Condition_time", "H4KO_0", "Cas9_0")) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id_version") %>% 
  left_join(id_symbol, by = "gene_id_version") %>% 
  dplyr::filter(log2FoldChange < -0.75 & padj < 0.05)
```

### extract the counts and save
```{r}
VSD <- assay(VST) |> 
  as.data.frame() |> 
  rownames_to_column('gene_id') |> 
  left_join(id_symbol, join_by('gene_id' == 'gene_id_version'))

VSD |> write_csv('VSD_normoxia_newsamples.csv')

DESeq2::counts(DDS, normalized = TRUE) |> 
  as.data.frame() |> 
  rownames_to_column('gene_id') |> 
  mutate(gene_id = str_remove(gene_id, "\\..*")) %>% 
  write_csv('Norm_counts_normoxia_newsamples.csv')
```

### save the upregulated and downregulated
```{r}
H4KO_Vs_Cas9_norm_Up %>% write_csv("Upreg_normoxia_H4KO_vs_ctrl_salmon.csv")
H4KO_Vs_Cas9_norm_Down %>% write_csv("Downreg_normoxia_H4KO_vs_ctrl_salmon.csv")
```

###############################################################
# Just to send emanuele the shared genes between the two clones
###############################################################


################################################
################################################

# Load the DBR and select only those on the promoters
```{r}
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene::TxDb.Hsapiens.UCSC.hg38.knownGene
library(org.Hs.eg.db)

H4KO_vs_Cas9_norm <- list(
  H3K27ac_H4KO_vs_Cas9_Up = readPeakFile("../ChIPseq/DiffBind_bed_files/entires/H3K27ac_Up_H4KO_vs_Cas9_Normoxia_DiffBind_enriched_peaks.bed"),
  H2BK120ac_H4KO_vs_Cas9_Up = readPeakFile("../ChIPseq/DiffBind_bed_files/entires/H2BK120ac_Up_H4KO_vs_Cas9_Normoxia_DiffBind_enriched_peaks.bed"),
  H3K27ac_H4KO_vs_Cas9_Down = readPeakFile("../ChIPseq/DiffBind_bed_files/entires/H3K27ac_Down_H4KO_vs_Cas9_Normoxia_DiffBind_enriched_peaks.bed"),
  H2BK120ac_H4KO_vs_Cas9_Down = readPeakFile("../ChIPseq/DiffBind_bed_files/entires/H2BK120ac_Down_H4KO_vs_Cas9_Normoxia_DiffBind_enriched_peaks.bed")
)

H4KO_vs_Cas9_norm_anno_promoter <- lapply(H4KO_vs_Cas9_norm, function(peaks) {
  Anno <- annotatePeak(peaks, tssRegion = c(-3000,3000), TxDb = txdb, annoDb = "org.Hs.eg.db", level = "transcript") %>% 
    as.data.frame() %>% 
    dplyr::filter(str_detect(annotation, "Promoter")) 
    return(Anno)
})
```


# Gene ontology enrichment
```{r}
GO_H4KO_vs_Cas9_norm_anno_promoter <- lapply(H4KO_vs_Cas9_norm_anno_promoter , function(df) {
  GO <- enrichGO(gene = df$SYMBOL, keyType = "SYMBOL", OrgDb = "org.Hs.eg.db", ont = "ALL")
  return(GO)
})

GO_Plot_list <- list()

for (name in names(GO_H4KO_vs_Cas9_norm_anno_promoter)) {
  plot <- dotplot(GO_H4KO_vs_Cas9_norm_anno_promoter[[name]], title = str_c(name,"\nGO Promoter (ALL)"))
  print(plot)
  ggsave(plot, filename=str_c("GO_Promoter_all_",name,".png"), height = 8, width = 8)
}

```

# Save excel to send Prof
```{r}
for (i in names(H4KO_vs_Cas9_norm_anno_promoter)) {
  openxlsx::write.xlsx(H4KO_vs_Cas9_norm_anno_promoter[[i]] %>% dplyr::select(seqnames, start, end, annotation, SYMBOL), file = paste0(i,".xlsx"))
}

```


# RNAseq file HDAC4 KO vs Cas9 Normoxia
# Comparison Degs RNAseq with Genes of DBRs in ChIPseq, DBR Genes linked with HiC, SEs genes linked with HiC

# Load DESeq2 Upreg
```{r}
Genes_UP_H4KO_vs_Cas9_norm <- read_csv("Upreg_normoxia_H4KO_vs_ctrl_salmon.csv")
Genes_UP_H4KO_vs_Cas9_norm <- na.omit(Genes_UP_H4KO_vs_Cas9_norm$symbol)


Genes_DOWN_H4KO_vs_Cas9_norm <- read_csv("Downreg_normoxia_H4KO_vs_ctrl_salmon.csv")
Genes_DOWN_H4KO_vs_Cas9_norm <- na.omit(Genes_DOWN_H4KO_vs_Cas9_norm$symbol)
```

# In alternative we can use the differentials of the overlap between the two clones
```{r}
Genes_UP_H4KO_vs_Cas9_norm <- openxlsx::read.xlsx("../RNAseq/Tab_files/Upreg_common_normoxia_newclones.xlsx", sheet = 1, colNames = F)
Genes_UP_H4KO_vs_Cas9_norm <- na.omit(Genes_UP_H4KO_vs_Cas9_norm$X1)
Genes_UP_H4KO_vs_Cas9_norm <- Genes_UP_H4KO_vs_Cas9_norm[Genes_UP_H4KO_vs_Cas9_norm != ""]

Genes_DOWN_H4KO_vs_Cas9_norm <- openxlsx::read.xlsx("../RNAseq/Tab_files/Upreg_common_normoxia_newclones.xlsx", sheet = 2, colNames = F)
Genes_DOWN_H4KO_vs_Cas9_norm <- na.omit(Genes_DOWN_H4KO_vs_Cas9_norm$X1)
Genes_DOWN_H4KO_vs_Cas9_norm <- Genes_DOWN_H4KO_vs_Cas9_norm[Genes_DOWN_H4KO_vs_Cas9_norm != ""]
```

# Or with the new clones 2025
```{r}
Genes_UP_H4KO_vs_Cas9_norm <- read_lines('../RNAseq2025/HDAC4_KO_vs_ctrl_normoxia_intersect_clones_sep.txt')
```


# DBRs genes
```{r}
Genes_DBR_H3K27ac_onPromoter <- H4KO_vs_Cas9_norm_anno_promoter$H3K27ac_H4KO_vs_Cas9_Up$SYMBOL %>% na.omit()
Genes_DBR_H2BK120ac_onPromoter <- H4KO_vs_Cas9_norm_anno_promoter$H2BK120ac_H4KO_vs_Cas9_Up$SYMBOL %>% na.omit()

sum(Genes_UP_H4KO_vs_Cas9_norm %in% Genes_DBR_H3K27ac_onPromoter)
sum(Genes_UP_H4KO_vs_Cas9_norm %in% Genes_DBR_H2BK120ac_onPromoter)

U <- VennDiagram::venn.diagram(x = list(Genes_UP_H4KO_vs_Cas9_norm,
                                   Genes_DBR_H3K27ac_onPromoter,
                                   Genes_DBR_H2BK120ac_onPromoter),
                          filename = NULL, output = T,
                          disable.logging = T,
                          category.names = c("Genes Up",
                                             "H3K27ac Genes Up Promoter",
                                             "H2BK120ac Genes Up Promoter"),
                          margin = 0.1)
grid::grid.newpage()
grid::grid.draw(U)



# Also Down
H3K27ac_Genes_DOWN_H4KO_vs_Cas9_norm_only_promoter <- H4KO_vs_Cas9_norm_anno_promoter$H3K27ac_H4KO_vs_Cas9_Down$SYMBOL %>% na.omit()
H2BK120ac_Genes_DOWN_H4KO_vs_Cas9_norm_only_promoter <- H4KO_vs_Cas9_norm_anno_promoter$H2BK120ac_H4KO_vs_Cas9_Down$SYMBOL %>% na.omit()

sum(Genes_DOWN_H4KO_vs_Cas9_norm %in% H3K27ac_Genes_DOWN_H4KO_vs_Cas9_norm_only_promoter )
sum(Genes_DOWN_H4KO_vs_Cas9_norm %in% H2BK120ac_Genes_DOWN_H4KO_vs_Cas9_norm_only_promoter)

D <- VennDiagram::venn.diagram(x = list(Genes_DOWN_H4KO_vs_Cas9_norm,
                                   H3K27ac_Genes_DOWN_H4KO_vs_Cas9_norm_only_promoter,
                                   H2BK120ac_Genes_DOWN_H4KO_vs_Cas9_norm_only_promoter),
                          filename = NULL, output = T,
                          disable.logging = T,
                          category.names = c("Genes Down",
                           "H3K27ac Genes Down",
                            "H2BK120ac Genes Down"),
                          margin = 0.1)

grid::grid.newpage()
grid::grid.draw(D)
```

# List of Genes to send Prof
```{r}
Genes_Up_shared_DBR_promoter <- Genes_UP_H4KO_vs_Cas9_norm[Genes_UP_H4KO_vs_Cas9_norm %in% Genes_DBR_H3K27ac_onPromoter & Genes_UP_H4KO_vs_Cas9_norm %in% Genes_DBR_H2BK120ac_onPromoter]

Genes_Up_shared_DBR_promoter
```


```{r}
tibble(Genes = Genes_Up_shared_DBR_promoter) %>%  write_csv(., file = "Genes_UP_RNA_H3_H2_H4KO_vs_Cas9.csv")
```

# HDAC4 on Core of genes Upregulated and with a Peak on the promoter of both H2BK120ac and H3K27ac
# I used bedtools intersect 
```{bash}
# Circa
bedtools intersect -a Gencode_V44_TSS.bed -b HDAC4.narrowpeak -u > Gencode_V44_TSS_withHDAC4.bed
```

```{r}
Genes_with_HDAC4_on_Promoter <- read_delim(file = "../Gencode_V44_TSS_withHDAC4.bed", col_names = F) %>% 
  dplyr::select(X4) %>% 
  dplyr::mutate(Genes = str_extract(X4, "(?<=gene_name )[^;]+"))

Genes_with_HDAC4_on_Promoter <- Genes_with_HDAC4_on_Promoter$Genes
Genes_with_HDAC4_on_Promoter <- unique(Genes_with_HDAC4_on_Promoter[!str_detect(Genes_with_HDAC4_on_Promoter, "ENSG00")])

X <- VennDiagram::venn.diagram(x = list(Genes_Up_shared_DBR_promoter, 
                                   Genes_with_HDAC4_on_Promoter),
                          filename = NULL, 
                          disable.logging = T,
                          category.names = c("68 genes core",
                                             "Genes with HDAC4 on Promoter"),
                          margin = 0.2, 
                          output = TRUE)

grid::grid.newpage()           # Start a fresh plotting page
grid::grid.draw(X)

print(Genes_Up_shared_DBR_promoter[Genes_Up_shared_DBR_promoter %in% Genes_with_HDAC4_on_Promoter])
```


# Genes contacted by SEs
```{r}
Genes_SEs_H3K27ac_HiC <- read_tsv("../HiC_data/SEs_H4KO_vs_Cas9_H3K27ac_Norm/genes_contacted.tsv", col_names = F)
Genes_SEs_H3K27ac_HiC <- na.omit(Genes_SEs_H3K27ac_HiC$X1)

Genes_SEs_H2BK120ac_HiC <- read_tsv("../HiC_data/SEs_H4KO_vs_Cas9_H2BK120ac_Norm/genes_contacted.tsv", col_names = F)
Genes_SEs_H2BK120ac_HiC <- na.omit(Genes_SEs_H2BK120ac_HiC$X1)
```

```{r}
X <- VennDiagram::venn.diagram(x = list(Genes_UP_H4KO_vs_Cas9_norm, 
                                   Genes_SEs_H3K27ac_HiC,
                                   Genes_SEs_H2BK120ac_HiC),
                          filename = NULL, 
                          disable.logging = T,
                          category.names = c("Genes Up",
                                             "SEs H3K27ac Genes HiC",
                                             "SEs H2Bk120ac Genes HiC"),
                          margin = 0.1,
                          output = TRUE)

grid::grid.newpage()           # Start a fresh plotting page
grid::grid.draw(X)

Genes_shared_all_SEs_HiC <- Genes_UP_H4KO_vs_Cas9_norm[Genes_UP_H4KO_vs_Cas9_norm %in% Genes_SEs_H3K27ac_HiC &
                            Genes_UP_H4KO_vs_Cas9_norm %in% Genes_SEs_H2BK120ac_HiC]
```

# I already did in the other script (Hic_genes_DBR.Rmd) a boxplot with the VSD
# Now we do the same as the SEs for the DBR in contact with genes
```{r}
library(ggVennDiagram)
Genes_DBR_H3K27ac_HiC <- read_tsv("../HiC_data/H3K27ac_Up_H4KO_vs_Cas9_Normoxia/genes_contacted.tsv", col_names = F)
Genes_DBR_H3K27ac_HiC <- na.omit(Genes_DBR_H3K27ac_HiC$X1)

Genes_DBR_H2BK120ac_HiC <- read_tsv("../HiC_data/H2BK120ac_Up_H4KO_vs_Cas9_Normoxia/genes_contacted.tsv", col_names = F)
Genes_DBR_H2BK120ac_HiC <- na.omit(Genes_DBR_H2BK120ac_HiC$X1)
```


```{r}
X <- VennDiagram::venn.diagram(x = list(Genes_UP_H4KO_vs_Cas9_norm, 
                                   Genes_DBR_H3K27ac_HiC,
                                   Genes_DBR_H2BK120ac_HiC),
                          filename = NULL, 
                          disable.logging = T,
                          category.names = c("Genes Up",
                                             "DBRs H3K27ac Genes HiC",
                                             "DBRs H2Bk120ac Genes HiC"),
                          margin = 0.1, 
                          output = TRUE)

grid::grid.newpage()           # Start a fresh plotting page
grid::grid.draw(X)

Genes_shared_all_TEs_HiC <- Genes_UP_H4KO_vs_Cas9_norm[Genes_UP_H4KO_vs_Cas9_norm %in% Genes_DBR_H3K27ac_HiC & 
                             Genes_UP_H4KO_vs_Cas9_norm %in% Genes_DBR_H2BK120ac_HiC]
```

# Save the list of genes
```{r}
write_lines(Genes_Up_shared_DBR_promoter, file = "Genes_Up_shared_DBR_promoter.txt")
write_lines(Genes_shared_all_TEs_HiC, file = "Genes_Up_shared_TEs_HiC.txt")
write_lines(Genes_shared_all_SEs_HiC, file = "Genes_Up_shared_SEs_HiC.txt")


# Genes_Up_H3K27ac_DBR_promoter <- Genes_UP_H4KO_vs_Cas9_norm[Genes_UP_H4KO_vs_Cas9_norm %in% Genes_DBR_H3K27ac_onPromoter]
# Genes_Up_H2BK120ac_DBR_promoter <- Genes_UP_H4KO_vs_Cas9_norm[Genes_UP_H4KO_vs_Cas9_norm %in% Genes_DBR_H2BK120ac_onPromoter]
# 
# write_lines(Genes_Up_H3K27ac_DBR_promoter, file="Genes_Up_H3K27ac_DBR_promoter.txt")
# write_lines(Genes_Up_H2BK120ac_DBR_promoter, file="Genes_Up_H2BK120ac_DBR_promoter.txt")

# Let's take also the genes that are only shared between H3K27ac and RNAseq and H2BK120AC and RNAseq
Genes_Up_DBR_promoter_both_marks_also_not_common <- unique(Genes_UP_H4KO_vs_Cas9_norm[Genes_UP_H4KO_vs_Cas9_norm %in% Genes_DBR_H3K27ac_onPromoter | Genes_UP_H4KO_vs_Cas9_norm %in% Genes_DBR_H2BK120ac_onPromoter])
write_lines(Genes_Up_DBR_promoter_both_marks_also_not_common, file="Genes_Up_H3K27ac_and_H2BK120ac_promoter.txt")

```

###############################################################
# Heatmap with signal on TSS and genes upregulated and HiC data
###############################################################

# Load the VSD counts and the DESeq results with separated clones
```{r}
VSD <- read_csv("VSD_normoxia_newsamples.csv") |> 
  dplyr::filter(!is.na(symbol) & !duplicated(symbol))

core_genes_list <- list(
  Genes_Up_shared_DBR_promoter = read_lines("Genes_Up_shared_DBR_promoter.txt"),
  Genes_Up_DBR_promoter_both_marks_also_not_common = read_lines("Genes_Up_H3K27ac_and_H2BK120ac_promoter.txt"),
  Genes_Up_shared_all_TEs_HiC = read_lines("Genes_Up_shared_TEs_HiC.txt"),
  Genes_Up_shared_all_SEs_HiC = read_lines("Genes_Up_shared_SEs_HiC.txt")  
)
```

```{r}
# filter the VSD on the core genes

filter_df <- function(df, genes) {
  
  df_f <- df %>% 
    dplyr::filter(symbol %in% genes) %>% 
    dplyr::select(-gene_id) %>% 
    column_to_rownames("symbol")
  
  return(df_f)
}

# Create VSD dataframe filtered on the gene lists
VSD_groups_list <- list()

for (i in names(core_genes_list)) {
  
  name = paste0("VSD_",i) %>% str_remove("Genes_")

  VSD_groups_list[[name]] <- filter_df(VSD, core_genes_list[[i]])
}
```

# Heatmap with signal of Acetylation
### load the file of the tss with bigwigs signal and apply log transformtion
```{r}
Table_Signal_Bw_TSS <- read_delim("../ChIPseq/TSS_bw_signal_bcoverage.bed") %>% 
  dplyr::mutate(symbol = str_extract(id, "(?<=gene_name )[^;]+")) |> 
  dplyr::filter(!is.na(symbol) & !duplicated(symbol)) %>% 
  dplyr::mutate(across(contains("RPKM"), log2))


```

# Load HDAC4 peaks and intersect with promoters, so adding the infos
# to the TSS file with Bigwig about the presence oh HDAC4
```{r}
HDAC4_peaks <- readPeakFile("../../NAR2020_Cb/GSM3882325_HDAC4_DMSO_SKUT_peaks.narrowPeak")

# Convert to Granges and create a new column with the counts of overlaps with HDAC4
Table_Signal_Bw_TSS_HDAC4 <- Table_Signal_Bw_TSS %>% 
  makeGRangesFromDataFrame(keep.extra.columns = T,
                           seqnames.field = "chr",
                           start.field = "start",
                           end.field = "end")
```

# Which of these core genes has a loop in which the opposite anchora has an overlap with HDAC4
# Can suggest a potential regulatory effect of HDAC4 also if there is no upregulated enhancer
# I don't know if this makes total sense but this is what the prof want
```{r}
# Load the chromatin loop file
chromatin_loop <- read_tsv("../HiC_data/chromatin_loops.bedpe", comment = "#", col_names = F)
chromatin_loop <- chromatin_loop[,1:6]

# I create 2 granges one on the anchor1 and one on the anchor2
anchor1 <- makeGRangesFromDataFrame(chromatin_loop,
                                    keep.extra.columns = T,
                                    seqnames.field = "X1",
                                    start.field = "X2",
                                    end.field = "X3")

names(mcols(anchor1)) <- c("1", "2", "3")

anchor2 <- makeGRangesFromDataFrame(chromatin_loop,
                                    keep.extra.columns = T,
                                    seqnames.field = "X4",
                                    start.field = "X5",
                                    end.field = "X6")

names(mcols(anchor2)) <- c("1", "2", "3")


# recrate the granges from the TSS table subsetted with the core genes
Table_Signal_Bw_TSS_HDAC4_core <- Table_Signal_Bw_TSS %>% 
  dplyr::filter(symbol %in% core_genes_list$Genes_Up_shared_DBR_promoter) %>% 
  makeGRangesFromDataFrame(keep.extra.columns = T,
                           seqnames.field = "chr",
                           start.field = "start",
                           end.field = "end")


###############
# This is hacky
###############

# create the granges with the rigth info inside
hits_1 <- findOverlaps(anchor1, Table_Signal_Bw_TSS_HDAC4_core)

# Extract query and subject indices
query_idx <- queryHits(hits_1)
subject_idx <- subjectHits(hits_1)

# Build new GRanges based on query ranges
gr_out_1 <- anchor1[query_idx]

# Add subject ranges as a metadata column
mcols(gr_out_1)$symbol <- Table_Signal_Bw_TSS_HDAC4_core$symbol[subject_idx]


# create the granges with the rigth info inside
hits_2 <- findOverlaps(anchor2, Table_Signal_Bw_TSS_HDAC4_core)

# Extract query and subject indices
query_idx <- queryHits(hits_2)
subject_idx <- subjectHits(hits_2)

# Build new GRanges based on query ranges
gr_out_2 <- anchor2[query_idx]

# Add subject ranges as a metadata column
mcols(gr_out_2)$symbol <- Table_Signal_Bw_TSS_HDAC4_core$symbol[subject_idx]


######################
# bind the two objects

# the coordinates in the metadata is the anchor that we want to see if it has HDAC4

Gr_out_merged <- c(gr_out_1, gr_out_2) %>% 
  as.data.frame()

genes_with_regulatory_anchor <- Gr_out_merged %>% 
  dplyr::select(!c(seqnames, start, end)) %>% 
  makeGRangesFromDataFrame(seqnames.field = "1", start.field = "2", end.field = "3", keep.extra.columns = T)

HDAC4_olaps_on_anchor <- tibble(
  symbol = genes_with_regulatory_anchor$symbol,
  HDAC4_on_distal_element = countOverlaps(genes_with_regulatory_anchor, HDAC4_peaks)
  ) %>% 
  group_by(symbol) %>% 
  summarize(HDAC4_on_distal_element = sum(HDAC4_on_distal_element))

```

# add the overlaps on the promoter with HDAC4
```{r}
Overlaps_HDAC4_TSS <- countOverlaps(Table_Signal_Bw_TSS_HDAC4, HDAC4_peaks)

Table_Signal_Bw_TSS_HDAC4 <- Table_Signal_Bw_TSS_HDAC4 %>% 
  as.data.frame() %>% 
  dplyr::mutate(HDAC4_counts = Overlaps_HDAC4_TSS) %>% 
  left_join(HDAC4_olaps_on_anchor, by = "symbol") %>% 
  dplyr::mutate(HDAC4_on_distal_element = ifelse(is.na(HDAC4_on_distal_element), 0, HDAC4_on_distal_element))
```


# Add also MEF2D signal
# Copied and pasted everything for HDAC4 and subs MEF2D
```{r}
MEF2D_peaks <- readPeakFile("../../NAR2020_Cb/GSM3882327_MEF2D_DMSO_SKUT_peaks.narrowPeak")

# Load the chromatin loop file
chromatin_loop <- read_tsv("../HiC_data/chromatin_loops.bedpe", comment = "#", col_names = F)
chromatin_loop <- chromatin_loop[,1:6]

# I create 2 granges one on the anchor1 and one on the anchor2
anchor1 <- makeGRangesFromDataFrame(chromatin_loop,
                                    keep.extra.columns = T,
                                    seqnames.field = "X1",
                                    start.field = "X2",
                                    end.field = "X3")

names(mcols(anchor1)) <- c("1", "2", "3")

anchor2 <- makeGRangesFromDataFrame(chromatin_loop,
                                    keep.extra.columns = T,
                                    seqnames.field = "X4",
                                    start.field = "X5",
                                    end.field = "X6")

names(mcols(anchor2)) <- c("1", "2", "3")


# recrate the granges from the TSS table subsetted with the core genes
Table_Signal_Bw_TSS_MEF2D_core <- Table_Signal_Bw_TSS %>% 
  dplyr::filter(symbol %in% core_genes_list$Genes_Up_shared_DBR_promoter) %>% 
  makeGRangesFromDataFrame(keep.extra.columns = T,
                           seqnames.field = "chr",
                           start.field = "start",
                           end.field = "end")


###############
# This is hacky
###############

# create the granges with the rigth info inside
hits_1 <- findOverlaps(anchor1, Table_Signal_Bw_TSS_MEF2D_core)

# Extract query and subject indices
query_idx <- queryHits(hits_1)
subject_idx <- subjectHits(hits_1)

# Build new GRanges based on query ranges
gr_out_1 <- anchor1[query_idx]

# Add subject ranges as a metadata column
mcols(gr_out_1)$symbol <- Table_Signal_Bw_TSS_MEF2D_core$symbol[subject_idx]


# create the granges with the rigth info inside
hits_2 <- findOverlaps(anchor2, Table_Signal_Bw_TSS_MEF2D_core)

# Extract query and subject indices
query_idx <- queryHits(hits_2)
subject_idx <- subjectHits(hits_2)

# Build new GRanges based on query ranges
gr_out_2 <- anchor2[query_idx]

# Add subject ranges as a metadata column
mcols(gr_out_2)$symbol <- Table_Signal_Bw_TSS_MEF2D_core$symbol[subject_idx]


######################
# bind the two objects

# the coordinates in the metadata is the anchor that we want to see if it has MEF2D

Gr_out_merged <- c(gr_out_1, gr_out_2) %>% 
  as.data.frame()

genes_with_regulatory_anchor <- Gr_out_merged %>% 
  dplyr::select(!c(seqnames, start, end)) %>% 
  makeGRangesFromDataFrame(seqnames.field = "1", start.field = "2", end.field = "3", keep.extra.columns = T)

MEF2D_olaps_on_anchor <- tibble(
  symbol = genes_with_regulatory_anchor$symbol,
  MEF2D_on_distal_element = countOverlaps(genes_with_regulatory_anchor, MEF2D_peaks)
  ) %>% 
  group_by(symbol) %>% 
  summarize(MEF2D_on_distal_element = sum(MEF2D_on_distal_element))


# Convert to Granges and create a new column with the counts of overlaps with MEF2D
Table_Signal_Bw_TSS_MEF2D <- Table_Signal_Bw_TSS %>% 
  makeGRangesFromDataFrame(keep.extra.columns = T,
                           seqnames.field = "chr",
                           start.field = "start",
                           end.field = "end")

Overlaps_MEF2D_TSS <- countOverlaps(Table_Signal_Bw_TSS_MEF2D, MEF2D_peaks)

Table_Signal_Bw_TSS_MEF2D <- Table_Signal_Bw_TSS_MEF2D %>% 
  as.data.frame() %>% 
  dplyr::mutate(MEF2D_counts = Overlaps_MEF2D_TSS) %>% 
  left_join(MEF2D_olaps_on_anchor, by = "symbol") %>% 
  dplyr::mutate(MEF2D_on_distal_element = ifelse(is.na(MEF2D_on_distal_element), 0, MEF2D_on_distal_element))
```

# Combine HDAC4 and MEF2D dataframe
```{r}
MEF2D_presence_df <- Table_Signal_Bw_TSS_MEF2D %>% dplyr::select(MEF2D_counts, MEF2D_on_distal_element)

HDAC4_MEF2D_acetylation_signal_df <- bind_cols(Table_Signal_Bw_TSS_HDAC4, MEF2D_presence_df)
```


# Create the dataframes for heatmap filtering on the gene lists
```{r}
filter_df <- function(df, genes) {
  df_f <- df %>% 
    dplyr::filter(symbol %in% genes) %>%       # use argument
    dplyr::select(c(symbol, contains("RPKM"), 
                    HDAC4_counts, HDAC4_on_distal_element, 
                    MEF2D_counts, MEF2D_on_distal_element)) %>% 
    column_to_rownames("symbol")
  return(df_f)
}

# Create Acetylation signal dataframe filtered on the gene lists
Acetylation_signal_list <- list()

for (i in names(core_genes_list)) {
  
  name = paste0("Acetylation_",i) %>% str_remove("Genes_")

  Acetylation_signal_list[[name]] <- filter_df(HDAC4_MEF2D_acetylation_signal_df, core_genes_list[[i]])
}
```

```{r}
df_Chip <- Acetylation_signal_list$Acetylation_Up_shared_DBR_promoter
df_Vsd <- VSD_groups_list$VSD_Up_shared_DBR_promoter

# df_Chip <- Acetylation_signal_list$Acetylation_Up_DBR_promoter_both_marks_also_not_common
# df_Vsd <- VSD_groups_list$VSD_Up_DBR_promoter_both_marks_also_not_common

# this for reorder correctly the rows
common_genes <- intersect(rownames(df_Vsd), rownames(df_Chip))
df_Vsd  <- df_Vsd[common_genes, , drop = FALSE]
df_Chip <- df_Chip[common_genes, , drop = FALSE]

par(bg=NA)

Core_Genes_Hmap_VSD_promoter <- Heatmap(matrix = df_Vsd,
        cluster_columns = F,
        name = "VSD",
        col = colorRamp2(colors = c("white", "red"), breaks = c(
          min(df_Vsd),
          max(df_Vsd)),
          ),
         rect_gp = gpar(col = "black", lwd = 0.5)
)

Acetylation_Hmap_Signal_promoter <- Heatmap(matrix = 
        df_Chip[, -c(9,10,11,12)], #HDAC4 counts is the 9 col
        cluster_columns = F,
        name = "log2 RPKM",
        col = colorRamp2(colors = c("white", "blue"), breaks = c(
          min(df_Chip[, -c(9,10,11,12)]),
          max(df_Chip[, -c(9,10,11,12)]))
          ),
        column_spl = c(rep("H3K27ac", 4), rep ("H2BK120ac", 4)),
        rect_gp = gpar(col = "black", lwd = 0.5)
        )

HDAC4_mini_Hmap <- Heatmap(matrix = 
        df_Chip[,9, drop = FALSE], #HDAC4 counts is the 9 col
        cluster_columns = F,
        name = "HDAC4 peaks",
        col = colorRamp2(colors = c("white", "pink", "purple"), 
                         breaks = c(0,1,2)),
        heatmap_legend_param = list(
              at = c(0, 1, 2)),
         rect_gp = gpar(col = "black", lwd = 0.5)
        )

HDAC4_distal_mini_Hmap <- Heatmap(matrix = 
        df_Chip[,10, drop = FALSE], #HDAC4 counts is the 9 col
        cluster_columns = F,
        name = "HDAC4 peaks on reg element",
        col = colorRamp2(colors = c("white", "pink", "purple"), 
                         breaks = c(0,1,2)),
        heatmap_legend_param = list(
              at = c(0, 1, 2)),
         rect_gp = gpar(col = "black", lwd = 0.5)
        )

MEF2D_mini_Hmap <- Heatmap(matrix = 
        df_Chip[,11, drop = FALSE], #HDAC4 counts is the 9 col
        cluster_columns = F,
        name = "MEF2D peaks",
        col = colorRamp2(colors = c("white", "lightgreen", "darkgreen"), 
                         breaks = c(0,1,2)),
        heatmap_legend_param = list(
              at = c(0, 1, 2)),
         rect_gp = gpar(col = "black", lwd = 0.5)
        )

MEF2D_distal_mini_Hmap <- Heatmap(matrix = 
        df_Chip[,12, drop = FALSE], #HDAC4 counts is the 9 col
        cluster_columns = F,
        name = "MEF2D peaks on reg element",
        col = colorRamp2(colors = c("white", "lightgreen", "darkgreen"), 
                         breaks = c(0,1,2)),
        heatmap_legend_param = list(
              at = c(0, 1, 2)),
         rect_gp = gpar(col = "black", lwd = 0.5)
        )


# Combined Heatmap
Hmap_promoter <- Core_Genes_Hmap_VSD_promoter + 
  Acetylation_Hmap_Signal_promoter + 
  HDAC4_mini_Hmap +
  HDAC4_distal_mini_Hmap +
  MEF2D_mini_Hmap +
  MEF2D_distal_mini_Hmap

png("Heatmap_Promoter.png", width = 22, height = 25, units = "cm", res = 600, bg = "transparent")
# png("Heatmap_Promoter_alsonotshared.png", width = 22, height = 40, units = "cm", res = 600, bg = "transparent")
draw(Hmap_promoter, background = 'white')
dev.off()

Hmap_promoter
```

# NFIB HDAC4 promoter motifs
```{r}
library(GenomicRanges)

NFIB_promoter <- Table_Signal_Bw_TSS %>% dplyr::filter(symbol == "NFIB") %>% makeGRangesFromDataFrame()

HDAC4_peak_on_NFIB <- subsetByOverlaps(HDAC4_peaks, NFIB_promoter)

# convert to bed
HDAC4_peak_on_NFIB %>% as.data.frame() %>% dplyr::select(seqnames, start, end) %>% write_delim('HDAC4_on_NFIB_promoter.bed', col_names = F, delim='\t')
```


# Let's take also the genes that are only shared between H3K27ac and RNAseq and H2BK120AC and RNAseq
```{r}
Genes_Up_DBR_promoter_both_marks_also_not_common <- unique(Genes_UP_H4KO_vs_Cas9_norm[Genes_UP_H4KO_vs_Cas9_norm %in% Genes_DBR_H3K27ac_onPromoter | Genes_UP_H4KO_vs_Cas9_norm %in% Genes_DBR_H2BK120ac_onPromoter])

Genes_Up_DBR_promoter_both_marks_also_not_common
```
