```{r setup, results = "hide"}
library(DiffBind)
library(tidyverse)
library(ggplot2)
library(ChIPseeker)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(clusterProfiler)
library(AnnotationHub)
library(org.Hs.eg.db)
library(xlsx)
library(GenomicRanges)
library(rtracklayer)
library(ChIPpeakAnno)
```


```{r}
coldata <- read.csv("/media/alessio/Data/hypoxia/ChIPseq/experiment.csv") |> 
  dplyr::select(!1:2) |> 
  dplyr::mutate(Tissue = "SKUT1") |> 
  dplyr::relocate(Tissue, .after = sampleID) |> 
  dplyr::rename(SampleID = sampleID) |> 
  dplyr::rename(Condition = IP) |> 
  dplyr::rename(Factor = line) |> 
  dplyr::relocate(Factor, .before = Condition) |> 
  dplyr::mutate(Treatment = case_when(
    str_detect(ControlID, "N") ~ "0h",
    str_detect(ControlID, "H6") ~ "6h",
    str_detect(ControlID, "H24") ~ "24h"
  )) |> 
  dplyr::relocate(Treatment, .after = Condition) |> 
  dplyr::mutate(Replicate = case_when(
    str_detect(SampleID, "C1[A-Za-z]") ~ "1",
    str_detect(SampleID, "C2[A-Za-z]") ~ "2",
    str_detect(SampleID, "K125") ~"1",
    str_detect(SampleID, "K26") ~ "2"
  )) |> 
  dplyr::relocate(Replicate, .after= Treatment) 
  
coldata <- coldata |> 
  dplyr::select(!bamReads)

coldata <- coldata |> 
  mutate(bamReads = paste0("/media/alessio/Data/hypoxia/ChIPseq/samples/",SampleID,"/human_only_sorted.bam")) |> 
  relocate(bamReads, .after = Replicate)

coldata <- coldata |> 
    dplyr::mutate(bamReads = sub("/[^/]*$", "", bamControl)) |> 
  dplyr::mutate(bamReads = paste0(bamReads,"/human_only_sorted.bam")) |> 
  dplyr::relocate(bamReads, .after = Replicate)


coldata = read.csv("/media/alessio/Data/hypoxia/ChIPseq/experiment.csv") |> 
  dplyr::mutate(bamReads = paste0("/media/alessio/Data/hypoxia/ChIPseq/samples/",Condition,"/",SampleID,"/","human_only_sorted.bam")) |> 
  dplyr::mutate(Peaks = paste0("/media/alessio/Data/hypoxia/ChIPseq/samples/",Condition,"/",SampleID,"/",SampleID,"_peaks.narrowPeak"))

write.csv(coldata, "/media/alessio/Data/hypoxia/ChIPseq/experiment.csv", row.names = F)

coldata = read.csv("/media/alessio/Data/hypoxia/ChIPseq/experiment.csv") |> 
  dplyr::mutate(Tissue = Factor) |> 
  dplyr::mutate(Factor = paste0(Factor,"_",Treatment))


coldata <- read.csv("/media/alessio/Data/hypoxia/ChIPseq/experiment.csv")

coldata <- bind_rows(coldata, (coldata |> dplyr::mutate(PeakCaller = "HOMER")))

coldata <- coldata |> 
  dplyr::mutate(Peaks = ifelse(PeakCaller == "HOMER", paste0("/media/alessio/Data/hypoxia/ChIPseq/Homer_anno/Tag_Directories/",SampleID,"_TagDir/","peaks.bed"), Peaks))


  
```

#Diffbind H3K27AC
#preprocessing dba count MACS3
```{r}
#leggo samplesheet
coldata = read.csv("/media/alessio/Data/hypoxia/ChIPseq/experiment.csv") |> 
  dplyr::filter(Condition == "H3K27AC" & PeakCaller == "macs3")

coldata <- coldata %>% 
  dplyr::mutate(Peaks = str_c("/media/alessio/Data/hypoxia/ChIPseq/macs3_H3K27ac/",SampleID,"_peaks.narrowPeak"))

ChIP <- dba(sampleSheet=coldata, scoreCol = 5)

plot(ChIP)

ChIP <- dba.count(ChIP, bUseSummarizeOverlaps = TRUE, bParallel = FALSE)
```


#preprocessing dba count HOMER
```{r}
#leggo samplesheet
coldata = read.csv("/media/alessio/Data/hypoxia/ChIPseq/experiment.csv") |> 
  dplyr::filter(Condition == "H3K27AC" & PeakCaller == "HOMER")


ChIP <- dba(sampleSheet=coldata, scoreCol = 5)

plot(ChIP)

ChIP <- dba.count(ChIP, bUseSummarizeOverlaps = TRUE, bParallel = FALSE)
```


#normalization
```{r}
ChIP_normalized <- dba.normalize(ChIP, normalize="RLE", library = DBA_LIBSIZE_PEAKREADS)

#provare dba_norm_lib

#png(filename = "/media/alessio/Data/hypoxia/ChIPseq/Figure_paper/H3K27ac_DiffBind_PCA.png", res = 300, units = "cm", height = 18, width = 18)
PCA_diffbind <- dba.plotPCA(ChIP_normalized, alpha = 0.7, attributes = DBA_TISSUE, label=DBA_ID)
#PCA_diffbind <- dba.plotPCA(ChIP_normalized, alpha = 0.7, attributes = DBA_TISSUE)
#dev.off()

#png(filename = "/media/alessio/Data/hypoxia/ChIPseq/plot/H3K27ac_DiffBind_SDist.png", res = 300, units = "cm", height = 18, width = 18)
SDIST <- plot(ChIP_normalized)
#dev.off()
```

#estraggo gli overlaps (voglio vedere se sono simili a quelli di bedtools) VENN DIAGRAMS su tutti i masks
```{r}
for (i in names(ChIP_normalized$masks)) {
   tryCatch({
    #png(filename = paste0("/media/alessio/Data/hypoxia/ChIPseq/plot/DiffBind_Venn/","DiffBind_Venn_",i,".png"),
       # units = "cm", width = 15, height = 15, res = 300)
    Venn <- dba.plotVenn(ChIP_normalized, ChIP_normalized$masks[[i]], main = paste0(i," consensus peakset DiffBind"))
    #dev.off()
   }, error = function(e) {
      cat("Error:", conditionMessage(e), "\n")
    # Continue to the next iteration
  })
}

rm(height)
```

#results
```{r}
#setting del contrast
Diff_peaks_contrast <- list(Normoxia_H4KO_vs_CAS9 = (Normoxia_H4KO_vs_CAS9 <- dba.contrast(ChIP_normalized,
                                                                                  design = ~ Factor, contrast = c("Factor", "H4KO_0h", "CAS9_0h"))),
                  H6_H4KO_vs_CAS9 = (H6_H4KO_vs_CAS9 <- dba.contrast(ChIP_normalized,
                                                                     design = ~ Factor, contrast = c("Factor", "H4KO_6h", "CAS9_6h"))),
                  H24_H4KO_vs_CAS9 = (H24_H4KO_vs_CAS9 <- dba.contrast(ChIP_normalized,
                                                                       design = ~ Factor, contrast = c("Factor", "H4KO_24h", "CAS9_24h")))
 
                                                      )
#calcolo dei picchi differenziali
Diff_peaks_analyze <- lapply(Diff_peaks_contrast, function(analyze) {
  analyzed <- dba.analyze(analyze)
  return(analyzed)
})

for (i in Diff_peaks_analyze) {
  show <- dba.show(i, bContrasts=TRUE)
  print(show)
  rm(i, show)
}
```

#info
```{r}
info <- dba.show(Diff_peaks_analyze$Normoxia_H4KO_vs_CAS9)
libsizes <- cbind(LibReads=info$Reads, FRiP=info$FRiP, PeakReads=round(info$Reads * info$FRiP))
rownames(libsizes) <- info$ID
libsizes

norm <- dba.normalize(ChIP_normalized, bRetrieve=TRUE)
norm
NORM_FACTORS <- (1/norm$norm.factors)
```

#Plot Venn, Heatmap, Volcano, Boxplot, MA
```{r}
#plotMA
for (i in Diff_peaks_analyze) {
  MA <- dba.plotMA(i, contrast = 1)
  print(MA)
  rm(i, MA)
}

#Venn
for (i in seq_along(Diff_peaks_analyze)) {
  namevenn <- names(Diff_peaks_analyze)[i]
  namevenn <- str_extract(namevenn, ("Normoxia|H\\d{1,2}"))
  #png(filename = paste0("/media/alessio/Data/hypoxia/ChIPseq/plot/","H3K27ac_Venn_Diffbind_",namevenn,".png"),
     # res = 300, units = "cm", height = 20, width = 20)
  Venn <- dba.plotVenn(Diff_peaks_analyze[[i]], contrast = 1, bDB = TRUE, bGain = TRUE, bLoss = TRUE, bAll = FALSE, bUsePval = T,
                       main = paste0("H3K27ac\n",namevenn))
  #dev.off()
}


#Volcano
for (i in seq_along(Diff_peaks_analyze)) {
  nameVolc <- names(Diff_peaks_analyze)[i]
  nameVolc <- str_extract(nameVolc, ("Normoxia|H\\d{1,2}"))
 # png(filename = paste0("/media/alessio/Data/hypoxia/ChIPseq/plot/","H3K27ac_Volcano_Diffbind_",nameVolc,".png"),
  #    res = 300, units = "cm", height = 20, width = 20)
  Volcano <- dba.plotVolcano(Diff_peaks_analyze[[i]],
                       bLabels = F, th = 0.05)
 # dev.off()
}


#Heatmap
hmap <- colorRampPalette(c("purple", "black", "green"))(n = 13)

for (i in seq_along(Diff_peaks_analyze)) {
  nameheat = names(Diff_peaks_analyze)[i]
  nameheat = str_extract(nameheat, ("Normoxia|H\\d{1,2}"))
  #png(filename = paste0("/media/alessio/Data/hypoxia/ChIPseq/plot/","H3K27ac_Hmap_Diffbind_",nameheat,".png"),
   #   res = 300, units = "cm", height = 20, width = 20)
  Heat <- dba.plotHeatmap(Diff_peaks_analyze[[i]], contrast = 1, correlations=FALSE, scale="row", colScheme = hmap,
                          main = str_c("H3K27ac HDAC4 KO vs control\n",nameheat))
#  dev.off()
}

#Boxplot
for (i in seq_along(Diff_peaks_analyze)) {
  nameBox = names(Diff_peaks_analyze)[i]
  nameBox= str_extract(nameBox, ("Normoxia|H\\d{1,2}"))
 # png(filename = paste0("/media/alessio/Data/hypoxia/ChIPseq/plot/","H3K27ac_Boxplot_Diffbind_",nameBox,".png"),
  #    res = 300, units = "cm", height = 20, width = 20)
  Box <- dba.plotBox(Diff_peaks_analyze[[i]], contrast = 1)
  #dev.off()
}
```



#report picchi differenziali
```{r}
#report dei picchi differenziali significativi
Diff_peaks_report_sign <- lapply(Diff_peaks_analyze, function(analyze) {
  report <- dba.report(analyze, bUsePval = T)
  return(report)
})

#report di tutti i picchi 
Diff_peaks_report_all <- lapply(Diff_peaks_analyze, function(analyze) {
  report <- dba.report(analyze, th = 1)
  return(report)
})

```

#ChIPseeker
#annotatepeak
```{r}
txdb = makeTxDbFromGFF("/media/alessio/Data/genome110/gencode/gencode.v44.annotation.gtf")
txdb = TxDb.Hsapiens.UCSC.hg38.knownGene

#leggo i picchi dal report di diffbind picchi significativi
Peaks_annotation_sign <- lapply(Diff_peaks_report_sign, function(annopeak) {
  annodf <- (
    annotatePeak(
      annopeak,
      tssRegion = c(-3000, 3000),
      TxDb = txdb,
      level = "gene",
      flankDistance = 5000,
      columns = c("ENTREZID", "ENSEMBL", "SYMBOL", "GENENAME"),
      annoDb = "org.Hs.eg.db"
    )
  )
  return(annodf)
})

#Sign peaks
for (i in seq_along(Peaks_annotation_sign)) {
  nameanno <- names(Peaks_annotation_sign)[i]
  #nameanno <- str_extract(nameanno, ("Normoxia|H\\d{1,2}"))
  TSS <- plotDistToTSS(Peaks_annotation_sign[[i]])
  print(TSS)
  annobar <- plotAnnoBar(Peaks_annotation_sign[[i]],
                         title = str_c("H3K27ac peak distribution ", nameanno))
  print(annobar)
}

#su tutti i picchi
Peaks_annotation_all <- lapply(Diff_peaks_report_all, function(annopeak) {
  annodf <- (
    annotatePeak(
      annopeak,
      tssRegion = c(-3000, 3000),
      TxDb = txdb,
      level = "gene",
      flankDistance = 5000,
      columns = c("ENTREZID", "ENSEMBL", "SYMBOL", "GENENAME"),
      annoDb = "org.Hs.eg.db"
    )
  )
  return(annodf)
})


#trasformo in dataframe per salvarlo a csv
Peaks_annotation_all_df <- lapply(Peaks_annotation_all, function(df) {
  dframe <- as.data.frame(df@anno)
  return(dframe)
})

#trasformo in dataframe per salvarlo a csv
Peaks_annotation_sign_df <- lapply(Peaks_annotation_sign, function(df) {
  dframe <- as.data.frame(df@anno)
  return(dframe)
})
```

#salvo in excel
```{r}
openxlsx::write.xlsx(Peaks_annotation_all_df, file = "/media/alessio/Data/hypoxia/ChIPseq/tables/H3K27ac_tables/H3K27ac_Diff_bind_peak_annotation_hg38.xlsx")
openxlsx::write.xlsx(Peaks_annotation_sign_df, file = "/media/alessio/Data/hypoxia/ChIPseq/tables/H3K27ac_tables/H3K27ac_Diff_bind_peak_annotation_significative_Differential_hg38.xlsx")
```


#ORA sui geni vicini ai picchi differenziali
```{r}
GO_enrich_sign_peaks <- lapply(Peaks_annotation_sign, function(ORA) {
  UP <- as.data.frame(ORA@anno) |> dplyr::filter(Fold > 0)
  enr <- enrichGO(gene = UP$SYMBOL,
                  OrgDb = org.Hs.eg.db,
                  keyType = "SYMBOL",
                  ont  = "ALL",
                  pAdjustMethod = "BH",
                  pvalueCutoff = 0.05,
                  qvalueCutoff = 0.2
  )
  return(enr)
})

for (i in names(GO_enrich_sign_peaks)) {
  name <- gsub("_", " ", i)
  dotplot <- dotplot(GO_enrich_sign_peaks[[i]], showCategory=20, color = "p.adjust") +
    labs(title = "GO enrichment",
         subtitle = name)
  #ggsave(plot = dotplot, filename = paste0("/media/alessio/Data/hypoxia/RNAseq/plot/ORA/","GO_enr_HDAC4_exclusive ",name,".png"), dpi = 300, height = 12, width = 9)
  print(dotplot)
}
```

#profili
```{r}
profiles <- dba.plotProfile(Diff_peaks_analyze$Normoxia_H4KO_vs_CAS9)
dba.plotProfile(profiles)
```


#carico i GTF filtrati su i geni UP e down regolati di RNASeq
```{r}
import_GTF <- function(path) {
  tmp <- read.delim(path)
  GRanges_obj <- GRanges(tmp[,1],
                         IRanges(tmp[,2], tmp[,3]),
                         Transcirpt = tmp[,4], 
                         strand = tmp[,6])
  return(GRanges_obj)
}


path = "/media/alessio/Data/hypoxia/ChIPseq/GTF_filtered_RNAseq/"

Upreg_beds <- c("Upreg_normoxia_H4KO_vs_ctrl.bed",
                "Upreg_6H_H4KO_vs_ctrl.bed", 
                "Upreg_24H_H4KO_vs_ctrl.bed", 
                "Upreg_48H_H4KO_vs_ctrl.bed")

Downreg_beds <- c("Downreg_normoxia_H4KO_vs_ctrl.bed",
                  "Downreg_6H_H4KO_vs_ctrl.bed", 
                  "Downreg_24H_H4KO_vs_ctrl.bed", 
                  "Downreg_48H_H4KO_vs_ctrl.bed")


Upreg_beds_list <- list()

for (i in Upreg_beds) {
  pupi <- file.path(path,i)
  tmp <- import_GTF(pupi)
  Upreg_beds_list[[i]] <- tmp
}

Downreg_beds_list <- list()

for (i in Downreg_beds) {
  pupi <- file.path(path,i)
  tmp <- import_GTF(pupi)
  Downreg_beds_list[[i]] <- tmp
}
```

#rifaccio i profili su i geni Up e Down di RNAseq (porca troia!)
```{r}
Prova_Normoxia <- dba.plotProfile(ChIP_normalized, sites = Upreg_beds_list$Upreg_normoxia_H4KO_vs_ctrl.bed)
dba.plotProfile(Prova_Normoxia)
#viene molto male e non penso che venga definito il TSS
```



#faccio un analisi/report dei picchi di macs3 e Homer
```{r}

#MACS3
paths <- coldata$Peaks

H3K27ac_macs3_peakFiles <- list()

for (i in paths) {
  name <- str_extract(i, "[^/]+$")
  name <- str_remove(name, "_peaks.narrowPeak")
  tmp_df <- read.delim(i, header = F)
  colnames(tmp_df) <- c("Chromosome", "Start", "End", "Peak_ID", "Peak_Score", "Strand", "FC_Peak_summit", "-log10(P-value)", "-log10(Q-value)", "Relative_Summit_Pos_to_Peak_Start")
  H3K27ac_macs3_peakFiles[[name]] <- tmp_df
}

H3K27ac_macs3_peakFiles_transformed <- lapply(H3K27ac_macs3_peakFiles, function(df) {
  dataf <- df |> 
    dplyr::mutate("P.value" = 10^(-(`-log10(P-value)`)), "Q.value" = 10^(-(`-log10(Q-value)`)))
return(dataf)
})

Reportino_Macs3 <- tibble()

for (i in names(H3K27ac_macs3_peakFiles_transformed)) {
  tmp_df <- data.frame(Sample = i,
                       Mean_Peak_Score = mean(H3K27ac_macs3_peakFiles_transformed[[i]]$Peak_Score),
                       Median_Peak_Score = median(H3K27ac_macs3_peakFiles_transformed[[i]]$Peak_Score),
                       Mean_Qvalue = mean(H3K27ac_macs3_peakFiles_transformed[[i]]$Q.value),
                       Mean_FC_Peak_summit = mean(H3K27ac_macs3_peakFiles_transformed[[i]]$FC_Peak_summit),
                       Peak_Count = nrow(H3K27ac_macs3_peakFiles_transformed[[i]]))
                       
  Reportino_Macs3 <- bind_rows(Reportino_Macs3, tmp_df)
    
}



#HOMER
paths_Homer <- "/media/alessio/Data/hypoxia/ChIPseq/Homer_anno/Tag_Directories"
dirs <- str_c(coldata$SampleID, "_TagDir")
file <- "peaks.txt"

H3K27ac_HOMER_peakFiles <- list()

for (i in dirs) {
  name <- str_remove(i, "_TagDir")
  tmp_path <- file.path(paths_Homer,i,file)
  tmp_df <- read.delim(tmp_path, skip = 37, header = T)
  tmp_df$p.value.vs.Control <- as.numeric( tmp_df$p.value.vs.Control)
  H3K27ac_HOMER_peakFiles[[name]] <- tmp_df
}

Reportino_tattico_HOMER <- tibble()

for (i in names(H3K27ac_HOMER_peakFiles)) {
  tmp_df <- data.frame(Sample = i,
                       Mean_findpeaks_score = mean(H3K27ac_HOMER_peakFiles[[i]]$findPeaks.Score),
                       Mean_FC = mean(H3K27ac_HOMER_peakFiles[[i]]$Fold.Change.vs.Control),
                       Peak_Count = nrow(H3K27ac_HOMER_peakFiles[[i]]),
                       Mean_pvalue = mean(H3K27ac_HOMER_peakFiles[[i]]$p.value.vs.Control))
  Reportino_tattico_HOMER <- bind_rows(Reportino_tattico_HOMER, tmp_df)
}

```

#trasformo tutto in GRanges e vedo overlap tra Homer e Macs3
```{r}
GRanges_H3K27ac_macs3_peakFiles_transformed <- lapply(H3K27ac_macs3_peakFiles_transformed, function(df) {
  GR <- makeGRangesFromDataFrame(df)
  return(GR)
})

GRanges_H3K27ac_HOMER_peakFiles <- lapply(H3K27ac_HOMER_peakFiles, function(df) {
  GR <- makeGRangesFromDataFrame(df)
  return(GR)
})

for (i in 1:12) {
  Venn <- makeVennDiagram(list(GRanges_H3K27ac_macs3_peakFiles_transformed[[i]], GRanges_H3K27ac_HOMER_peakFiles[[i]]),
                  NameOfPeaks = c(paste0("MACS3_",names(GRanges_H3K27ac_macs3_peakFiles_transformed[i])),
                                  paste0("HOMER_",names(GRanges_H3K27ac_HOMER_peakFiles[i]))),
                  fill=c("#D55E00", "#0072B2"),
                  by = "region")
  print(Venn)
}

GRanges_H3K27ac_macs3_peakFiles_transformed$K125G1

```

#vedo overlaps tra Diffbind differential peaks di H4KO in normossia e i picchi esclusivi di bedtools
```{r}
H4KO_normoxia_bedtools_1 <- readPeakFile("/media/alessio/Data/hypoxia/ChIPseq/samples/H3K27AC/K125A1/K125A1_peaks.narrowPeak", as = "GRanges")

H4KO_normoxia_bedtools_2 <- readPeakFile("/media/alessio/Data/hypoxia/ChIPseq/samples/H3K27AC/K26A1/K26A1_peaks.narrowPeak", as = "GRanges")

H4KO_normoxia_Diffbind <- makeGRangesFromDataFrame(df = (read.xlsx("/media/alessio/Data/hypoxia/ChIPseq/tables/H3K27ac_Diff_bind_peak_annotation_significative_Differential.xlsx", sheetIndex = 1) |> 
  dplyr::filter(Fold > 0)))

makeVennDiagram(Peaks = list(H4KO_normoxia_bedtools_1, H4KO_normoxia_bedtools_2, H4KO_normoxia_Diffbind))

#i picchi di DiffBind si overlappano quasi interamente solo se prendo i replicati separati, se uso l'intersect di bedtools invece una buona parte viene esclusa, si vede che il metodo di bedtools e diffbind per creare un consensus di picchi Ã¨ diverso
```








#Analisi differenziale dei picchi hypoxia vs normoxia con le linee separate, (vedo effetto indotto da ipossia)

```{r}
Vs_normoxia_Diff_peaks_contrast <- list(
  CAS9_6h = (CAS9_6h <- dba.contrast(ChIP_normalized, design = ~ Factor, contrast = c("Factor", "CAS9_6h", "CAS9_0h"))),
  CAS9_24h = (CAS9_24h <- dba.contrast(ChIP_normalized, design = ~ Factor, contrast = c("Factor", "CAS9_24h", "CAS9_0h"))),
  H4KO_6h = (H4KO_6h <- dba.contrast(ChIP_normalized, design = ~ Factor, contrast = c("Factor", "H4KO_6h", "H4KO_0h"))),
  H4KO_24h = (H4KO_24h <- dba.contrast(ChIP_normalized, design = ~ Factor, contrast = c("Factor", "H4KO_24h", "H4KO_0h")))
 )

#calcolo dei picchi differenziali
Vs_normoxia_analyzed_DiffPeaks <- lapply(Vs_normoxia_Diff_peaks_contrast, function(contrast) {
  analyzed <- dba.analyze(contrast)
  return(analyzed)
})

for (i in Vs_normoxia_analyzed_DiffPeaks) {
  show <- dba.show(i, bContrasts=TRUE)
  print(show)
  rm(i, show)
}

Vs_normoxia_report_DiffPeaks <- lapply(Vs_normoxia_analyzed_DiffPeaks, function(df) {
  report <- dba.report(df, bUsePval = T)
  return(report)
})

Vs_normoxia_report_DiffPeaks_total <- lapply(Vs_normoxia_analyzed_DiffPeaks, function(df) {
  report <- dba.report(df, th = 1)
  return(report)
})

```



#plots
```{r}
#plotMA
for (i in Vs_normoxia_analyzed_DiffPeaks) {
  MA <- dba.plotMA(i, contrast = 1, bUsePval = T)
  print(MA)
  rm(i, MA)
}

#Volcano
for (i in seq_along(Vs_normoxia_analyzed_DiffPeaks)) {
  nameVolc <- names(Vs_normoxia_analyzed_DiffPeaks)[i]
  #nameVolc <- str_extract(nameVolc, ("Normoxia|H\\d{1,2}"))
  #png(filename = paste0("/media/alessio/Data/hypoxia/ChIPseq/plot/","H3K27ac_Vs_normoxia",nameVolc,".png"),
      #res = 300, units = "cm", height = 20, width = 20)
  Volcano <- dba.plotVolcano(Vs_normoxia_analyzed_DiffPeaks[[i]],
                       bLabels = F, th = 0.05)
  #dev.off()
}

#Heatmap
hmap <- colorRampPalette(c("purple", "black", "green"))(n = 13)
dba.plotHeatmap(Vs_normoxia_analyzed_DiffPeaks$CAS9_24h, contrast = 1, correlations=FALSE, scale="row", colScheme = hmap)
dba.plotHeatmap(Vs_normoxia_analyzed_DiffPeaks$H4KO_24h, contrast = 1, correlations=FALSE, scale="row", colScheme = hmap)
```

#Annoto i geni
```{r}
txdb =  makeTxDbFromGFF("/media/alessio/Data/genome110/gencode/gencode.v44.annotation.gtf")
txdb = TxDb.Hsapiens.UCSC.hg38.knownGene

Annotation_Vs_normoxia_DiffPeaks <- lapply(Vs_normoxia_report_DiffPeaks, function(annopeak) {
  annodf <- (annotatePeak( annopeak,
                           tssRegion = c(-3000, 3000),
                           TxDb = txdb,
                           level = "gene",
                           flankDistance = 5000,
                           columns = c("ENTREZID", "ENSEMBL", "SYMBOL", "GENENAME"),
                           annoDb = "org.Hs.eg.db"))
  return(annodf)
})

#trasformo in dataframe per salvarlo a csv
Annotation_Vs_normoxia_DiffPeaks_df <- lapply(Annotation_Vs_normoxia_DiffPeaks, function(df) {
  dframe <- as.data.frame(df@anno)
  return(dframe)
})

Annotation_Vs_normoxia_DiffPeaks_total <- lapply(Vs_normoxia_report_DiffPeaks_total, function(annopeak) {
  annodf <- (annotatePeak( annopeak,
                           tssRegion = c(-3000, 3000),
                           TxDb = txdb,
                           level = "gene",
                           flankDistance = 5000,
                           columns = c("ENTREZID", "ENSEMBL", "SYMBOL", "GENENAME"),
                           annoDb = "org.Hs.eg.db"))
  return(annodf)
})

#trasformo in dataframe per salvarlo a csv
Annotation_Vs_normoxia_DiffPeaks_total_df <- lapply(Annotation_Vs_normoxia_DiffPeaks_total, function(df) {
  dframe <- as.data.frame(df@anno)
  return(dframe)
})
```

#salvo in excel
```{r}
openxlsx::write.xlsx(Annotation_Vs_normoxia_DiffPeaks_df, file = "/media/alessio/Data/hypoxia/ChIPseq/tables/H3K27ac_tables/Vs_normoxia/H3K27ac_Vs_normoxia_annotation_significative_Differential_hg38.xlsx")
openxlsx::write.xlsx(Annotation_Vs_normoxia_DiffPeaks_total_df, file = "/media/alessio/Data/hypoxia/ChIPseq/tables/H3K27ac_tables/Vs_normoxia/H3K27ac_Vs_normoxia_annotation_total_hg38.xlsx")
```

#Go enrichment
```{r}
sheets <- readxl::excel_sheets("/media/alessio/Data/hypoxia/ChIPseq/tables/H3K27ac_tables/H3K27ac_Vs_normoxia_annotation_significative_Differential.xlsx")
Annotation_Vs_normoxia_DiffPeaks_df <- read_excel_allsheets("/media/alessio/Data/hypoxia/ChIPseq/tables/H3K27ac_tables/H3K27ac_Vs_normoxia_annotation_significative_Differential.xlsx")

GO_Annotation_Vs_normoxia_DiffPeaks_df <- lapply(Annotation_Vs_normoxia_DiffPeaks_df, function(genes) {
 genes <- genes |> 
   dplyr::filter(Fold > 0) |> 
   dplyr::select(SYMBOL) |> unlist()
 GO <- enrichGO(gene = genes,
                   OrgDb = org.Hs.eg.db,
                    keyType = "SYMBOL",
                         ont  = "ALL",
                         pAdjustMethod = "BH",
                         qvalueCutoff  = 0.5,
                 pvalueCutoff = 0.1
                        )
  return(GO)
})

for (i in names(GO_Annotation_Vs_normoxia_DiffPeaks_df)) {
  dotplot <- dotplot(GO_Annotation_Vs_normoxia_DiffPeaks_df[[i]], showCategory = 20) + ggtitle(i)
  print(dotplot)
}
```


#Overlapping regions tra CAS9 e H4KO a 24 ore
```{r}
Venn.diagram.UP.Peaks <- function(GR1, GR2) {
  UP.CAS9 <- as.data.frame(GR1) |> 
    dplyr::filter(Fold > 0)
  UP.CAS9.Granges <- makeGRangesFromDataFrame(UP.CAS9)
  
    UP.H4KO <- as.data.frame(GR2) |> 
    dplyr::filter(Fold > 0)
  UP.H4KO.Granges <- makeGRangesFromDataFrame(UP.H4KO)
  
  Venn <- makeVennDiagram(Peaks = list(UP.CAS9.Granges, UP.H4KO.Granges),
                          NameOfPeaks = c("UP Cas9 24h", "UP H4KO 24h"),
                          margin = 0.1)
  print(Venn)
  
}

png(filename = "/media/alessio/Data/hypoxia/ChIPseq/plot/Vs_normoxia_intersection_24h.png", units = "cm", height = 10, width = 13, res = 300)
Venn.diagram.UP.Peaks(Vs_normoxia_report_DiffPeaks$CAS9_24h, Vs_normoxia_report_DiffPeaks$H4KO_24h)
dev.off()
```




#Voglio vedere l'overlap dei picchi con HDAC4
#Carico picchi di DiffBind dai fogli excel che avevo creato in precedenza 
```{r}
sheets <- readxl::excel_sheets("/media/alessio/Data/hypoxia/ChIPseq/tables/H3K27ac_tables/H3K27ac_Diff_bind_peak_annotation.xlsx")

read_excel_allsheets <- function(filename, tibble = FALSE) {
    # I prefer straight data.frames
    # but if you like tidyverse tibbles (the default with read_excel)
    # then just pass tibble = TRUE
    sheets <- readxl::excel_sheets(filename)
    x <- lapply(sheets, function(X) readxl::read_excel(filename, sheet = X))
    if(!tibble) x <- lapply(x, as.data.frame)
    names(x) <- sheets
    x
}

DiffBind.peaks.H4KOvsCAS9 <- read_excel_allsheets("/media/alessio/Data/hypoxia/ChIPseq/tables/H3K27ac_tables/H3K27ac_Diff_bind_peak_annotation.xlsx")
```

#separo Up e Down regolati e Trasformo in Granges
```{r}
DiffBind.peaks.H4KOvsCAS9.UP.DOWN.GRanges <- list()

for (i in names(DiffBind.peaks.H4KOvsCAS9)) {
  dfUP <- DiffBind.peaks.H4KOvsCAS9[[i]] |> 
    dplyr::filter(Fold > 0 & p.value <= 0.05)
  dfDOWN <- DiffBind.peaks.H4KOvsCAS9[[i]] |> 
    dplyr::filter(Fold < 0 & p.value <= 0.05)
  
  DiffBind.peaks.H4KOvsCAS9.UP.DOWN.GRanges[[(paste0("UP_",i))]] <- makeGRangesFromDataFrame(dfUP, keep.extra.columns = T) 
  DiffBind.peaks.H4KOvsCAS9.UP.DOWN.GRanges[[(paste0("DOWN_",i))]] <- makeGRangesFromDataFrame(dfDOWN, keep.extra.columns = T) 
}
```

#funzione temporanea per averlo come dataframe invece di GRANGES
```{r}
DiffBind.peaks.H4KOvsCAS9.UP.DOWN.DF <- list()

for (i in names(DiffBind.peaks.H4KOvsCAS9)) {
  dfUP <- DiffBind.peaks.H4KOvsCAS9[[i]] |> 
    dplyr::filter(Fold > 0 & p.value <= 0.05)
  dfDOWN <- DiffBind.peaks.H4KOvsCAS9[[i]] |> 
    dplyr::filter(Fold < 0 & p.value <= 0.05)
  
  DiffBind.peaks.H4KOvsCAS9.UP.DOWN.DF[[(paste0("UP_",i))]] <- dfUP 
  DiffBind.peaks.H4KOvsCAS9.UP.DOWN.DF[[(paste0("DOWN_",i))]] <- dfDOWN
}
```


#carico bed di HDAC4 e calcolo overlaps
```{r, suppressWarnings = T}
HDAC4.ChIP <- readPeakFile("/media/alessio/Data/hypoxia/ChIPseq/ChIP_atlas_files/HDAC4.ChIP.SKUT1.WT.bed")

for (i in names(DiffBind.peaks.H4KOvsCAS9.UP.DOWN.GRanges)) {
  Venn <- makeVennDiagram(Peaks = list(DiffBind.peaks.H4KOvsCAS9.UP.DOWN.GRanges[[i]], HDAC4.ChIP), NameOfPeaks = c(i, "HDAC4.ChIP"))
  print(Venn)
}
```



