```{r setup, echo = F}
library(tidyverse)
library(DiffBind)
library(ChIPseeker)
library(ChIPpeakAnno)
library(GenomicFeatures)
library(clusterProfiler)
library(org.Hs.eg.db)
```

```{r}
IDR.files = rep(c("Cas9_normoxia.bed", "Cas9_H6.bed", "Cas9_H24.bed", "H4KO_normoxia.bed", "H4KO_H6.bed", "H4KO_H24.bed"), 2)

coldata.IDR <- read.csv("/media/alessio/Data/hypoxia/ChIPseq/experiment.csv") |> 
  dplyr::filter(Condition == "H3K27AC" & PeakCaller == "macs3") |> 
  dplyr::mutate(Peaks = paste0("/media/alessio/Data/hypoxia/ChIPseq/IDR/IDR_H3K27AC/idr_results/",IDR.files))

ChIP <- dba(sampleSheet=coldata.IDR)

plot(ChIP)

ChIP <- dba.count(ChIP, bParallel = FALSE)
```
#normalization
```{r}
ChIP_normalized <- dba.normalize(ChIP, normalize="RLE", library = DBA_LIBSIZE_PEAKREADS)

#provare dba_norm_lib

#png(filename = "/media/alessio/Data/hypoxia/ChIPseq/Figure_paper/H3K27ac_DiffBind_PCA.png", res = 300, units = "cm", height = 18, width = 18)
PCA_diffbind <- dba.plotPCA(ChIP_normalized, alpha = 0.7, attributes = DBA_TISSUE, label=DBA_ID)
#PCA_diffbind <- dba.plotPCA(ChIP_normalized, alpha = 0.7, attributes = DBA_TISSUE)
#dev.off()

#png(filename = "/media/alessio/Data/hypoxia/ChIPseq/plot/H3K27ac_DiffBind_SDist.png", res = 300, units = "cm", height = 18, width = 18)
SDIST <- plot(ChIP_normalized)
#dev.off()
```


#results
```{r}
#setting del contrast
Diff_peaks_contrast <- list(Normoxia_H4KO_vs_CAS9 = (Normoxia_H4KO_vs_CAS9 <- dba.contrast(ChIP_normalized,
                                                                                  design = ~ Factor, contrast = c("Factor", "H4KO_0h", "CAS9_0h"))),
                  H6_H4KO_vs_CAS9 = (H6_H4KO_vs_CAS9 <- dba.contrast(ChIP_normalized,
                                                                     design = ~ Factor, contrast = c("Factor", "H4KO_6h", "CAS9_6h"))),
                  H24_H4KO_vs_CAS9 = (H24_H4KO_vs_CAS9 <- dba.contrast(ChIP_normalized,
                                                                       design = ~ Factor, contrast = c("Factor", "H4KO_24h", "CAS9_24h")))
 
                                                      )
#calcolo dei picchi differenziali
Diff_peaks_analyze <- lapply(Diff_peaks_contrast, function(analyze) {
  analyzed <- dba.analyze(analyze)
  return(analyzed)
})

for (i in Diff_peaks_analyze) {
  show <- dba.show(i, bContrasts=TRUE)
  print(show)
  rm(i, show)
}
```


#Plot Venn, Heatmap, Volcano, Boxplot, MA
```{r}
#plotMA
for (i in Diff_peaks_analyze) {
  MA <- dba.plotMA(i, contrast = 1)
  print(MA)
  rm(i, MA)
}

#Venn
for (i in seq_along(Diff_peaks_analyze)) {
  namevenn <- names(Diff_peaks_analyze)[i]
  namevenn <- str_extract(namevenn, ("Normoxia|H\\d{1,2}"))
  #png(filename = paste0("/media/alessio/Data/hypoxia/ChIPseq/plot/","H3K27ac_Venn_Diffbind_",namevenn,".png"),
     # res = 300, units = "cm", height = 20, width = 20)
  Venn <- dba.plotVenn(Diff_peaks_analyze[[i]], contrast = 1, bDB = TRUE, bGain = TRUE, bLoss = TRUE, bAll = FALSE, bUsePval = T,
                       main = paste0("H3K27ac\n",namevenn))
  #dev.off()
}


#Volcano
for (i in seq_along(Diff_peaks_analyze)) {
  nameVolc <- names(Diff_peaks_analyze)[i]
  nameVolc <- str_extract(nameVolc, ("Normoxia|H\\d{1,2}"))
 # png(filename = paste0("/media/alessio/Data/hypoxia/ChIPseq/plot/","H3K27ac_Volcano_Diffbind_",nameVolc,".png"),
  #    res = 300, units = "cm", height = 20, width = 20)
  Volcano <- dba.plotVolcano(Diff_peaks_analyze[[i]],
                       bLabels = F, th = 0.05)
 # dev.off()
}


#Heatmap
hmap <- colorRampPalette(c("purple", "black", "green"))(n = 13)

for (i in seq_along(Diff_peaks_analyze)) {
  nameheat = names(Diff_peaks_analyze)[i]
  nameheat = str_extract(nameheat, ("Normoxia|H\\d{1,2}"))
  #png(filename = paste0("/media/alessio/Data/hypoxia/ChIPseq/plot/","H3K27ac_Hmap_Diffbind_",nameheat,".png"),
   #   res = 300, units = "cm", height = 20, width = 20)
  Heat <- dba.plotHeatmap(Diff_peaks_analyze[[i]], contrast = 1, correlations=FALSE, scale="row", colScheme = hmap,
                          main = str_c("H3K27ac HDAC4 KO vs control\n",nameheat))
#  dev.off()
}

#Boxplot
for (i in seq_along(Diff_peaks_analyze)) {
  nameBox = names(Diff_peaks_analyze)[i]
  nameBox= str_extract(nameBox, ("Normoxia|H\\d{1,2}"))
 # png(filename = paste0("/media/alessio/Data/hypoxia/ChIPseq/plot/","H3K27ac_Boxplot_Diffbind_",nameBox,".png"),
  #    res = 300, units = "cm", height = 20, width = 20)
  Box <- dba.plotBox(Diff_peaks_analyze[[i]], contrast = 1)
  #dev.off()
}
```



#report picchi differenziali
```{r}
#report dei picchi differenziali significativi
Diff_peaks_report_sign <- lapply(Diff_peaks_analyze, function(analyze) {
  report <- dba.report(analyze, bUsePval = T)
  return(report)
})

#report di tutti i picchi 
Diff_peaks_report_all <- lapply(Diff_peaks_analyze, function(analyze) {
  report <- dba.report(analyze, th = 1)
  return(report)
})
```

#ChIPseeker
#annotatepeak
```{r}
txdb = makeTxDbFromGFF("/media/alessio/Data/genome110/gencode/gencode_V44.gff")

#leggo i picchi dal report di diffbind picchi significativi
Peaks_annotation_sign <- lapply(Diff_peaks_report_sign, function(annopeak) {
  annodf <- (annotatePeak(annopeak,
                          columns = c("ENTREZID", "ENSEMBL", "SYMBOL", "GENENAME"),
                          annoDb="org.Hs.eg.db",
                          TxDb = txdb, tssRegion=c(-3000, 3000)))
  return(annodf)
})

#Sign peaks
for (i in seq_along(Peaks_annotation_sign)) {
  nameanno <- names(Peaks_annotation_sign)[i]
  #nameanno <- str_extract(nameanno, ("Normoxia|H\\d{1,2}"))
  annobar <- plotAnnoBar(Peaks_annotation_sign[[i]], title = str_c("H3K27ac peak distribution ",nameanno))
  print(annobar)
}

#su tutti i picchi
Peaks_annotation_all <- lapply(Diff_peaks_report_all, function(annopeak) {
  annodf <- (annotatePeak(annopeak,
                          columns = c("ENTREZID", "ENSEMBL", "SYMBOL", "GENENAME"),
                          annoDb="org.Hs.eg.db",
                          TxDb = txdb, tssRegion=c(-3000, 3000)))
  return(annodf)
})

#trasformo in dataframe per salvarlo a csv
Peaks_annotation_all_df <- lapply(Peaks_annotation_all, function(df) {
  dframe <- as.data.frame(df@anno)
  return(dframe)
})

#trasformo in dataframe per salvarlo a csv
Peaks_annotation_sign_df <- lapply(Peaks_annotation_sign, function(df) {
  dframe <- as.data.frame(df@anno)
  return(dframe)
})
```

```{r}
openxlsx::write.xlsx(Peaks_annotation_all_df, file = "/media/alessio/Data/hypoxia/ChIPseq/IDR_DiffBind_H3K27ac/IDR_H3K27ac_Diff_bind_peak_annotation.xlsx")
openxlsx::write.xlsx(Peaks_annotation_sign_df, file = "/media/alessio/Data/hypoxia/ChIPseq/IDR_DiffBind_H3K27ac/IDR_H3K27ac_Diff_bind_peak_annotation_significative_Differential.xlsx")
```

#Gene Ontology
```{r}
for (i in names(Peaks_annotation_sign_df)) {
  GO <- enrichGO(Peaks_annotation_sign_df[[i]] |> 
                   dplyr::filter(Fold > 0 ) |> 
                   dplyr::select(SYMBOL) |> unlist(),
                 keyType = "SYMBOL",
                 OrgDb = org.Hs.eg.db,
                 ont = "ALL")
  tryCatch({ dotplot <- dotplot(GO, showCategory = 20, title = i)
  print(dotplot)},
  error = function(e) {
      cat("Error:", conditionMessage(e), "\n")
    # Continue to the next iteration
  })
}
```



