```{r setup, echo = F}
library(tidyverse)
library(GenomicRanges)
library(ChIPseeker)
library(ChIPpeakAnno)
library(rtracklayer)
library(GenomicFeatures)
```

##########
#HIF1A
##########


#leggo tutti le ChIP di HIF1a (consensus fra replicati)
```{r}
HIF1a.Hutsmc = ("PRJNA809308/macs3_peaks/SRR18097068_macs3_peaks.narrowPeak")
HIF1a.4.cell.lines = list.files("PRJNA823676/Consensus_bed", pattern = "HIF1A", full.names = T)

HIF1a.all.files <- c(HIF1a.Hutsmc, HIF1a.4.cell.lines)

HIF1a.all.Granges <- list()

for (i in HIF1a.all.files) {
  
  name = basename(i)
  
  tmp.Gr <- readPeakFile(i)
  
  HIF1a.all.Granges[[name]] <- tmp.Gr
}


 # mcols(HIF1a.all.Granges[[1]])$X = "ciao"

summary(HIF1a.all.Granges)
```

```{r}
png("Venn_all_HIF1a_cell_lines.png", res = 300, height = 22, width = 22, units = "cm")
makeVennDiagram(Peaks = list(HIF1a.all.Granges[[1]], HIF1a.all.Granges[[2]], HIF1a.all.Granges[[3]], HIF1a.all.Granges[[4]], HIF1a.all.Granges[[5]]), maxgap = 400,
              NameOfPeaks = names(HIF1a.all.Granges), margin =0.2)
dev.off()
```

#Annoto con ChIPseeker
```{r}
#txdb = TxDb.Hsapiens.UCSC.hg38.knownGene::TxDb.Hsapiens.UCSC.hg38.knownGene
txdb = makeTxDbFromGFF("/media/alessio/Data/genome110/gencode/gencode.v44.annotation.gtf")

HIF1a.summed.Anno <- c(HIF1a.all.Granges[[1]], HIF1a.all.Granges[[2]], HIF1a.all.Granges[[3]], HIF1a.all.Granges[[4]], HIF1a.all.Granges[[5]]) %>% 
  GenomicRanges::reduce(.) %>% 
  annotatePeak(TxDb = txdb,
                annoDb = "org.Hs.eg.db",
                tssRegion = c(-3000, 3000),
                flankDistance = 5000,
                level = "gene")

enrichGO(HIF1a.summed.Anno@anno %>% as.data.frame() %>% 
           dplyr::filter(str_detect(annotation, "Promoter")) %>% 
           dplyr::select(SYMBOL) %>%  unlist(),
         OrgDb = "org.Hs.eg.db",
         keyType = "SYMBOL",
         ont = "BP") %>% 
  dotplot(.) %>% 
  ggsave(, filename = "HIF1a_5_cell_lines_GO_enrichment_GTfV44_only_promoter.png", dpi = 300, height = 8, width = 7)

(plotAnnoBar(HIF1a.summed.Anno) + scale_fill_viridis_d()) %>% ggsave(., filename = "HIF1a_5_cell_lines_Peak_distribution.png", dpi = 300, height = 4, width = 7) 
#viene simile con entrambe le annotazioni
```

#salvo lista di geni in cui Ã¨ presente un picco di HIF1a sul promotore
```{r}
HIF1a.summed.Anno@anno %>% 
  as.data.frame() %>% 
  dplyr::filter(str_detect(annotation, "Promoter")) %>% 
  dplyr::select(SYMBOL) %>% 
  unlist() %>%
  na.omit() %>% 
  unique() %>% 
  write_lines(., "HIF1a_5_celllines_peaks_in_promoter_genes_GTFv44.txt")
```

#metto tutti insieme in un unico file
```{r}
HIF1a.summed <- c(HIF1a.all.Granges[[1]], HIF1a.all.Granges[[2]], HIF1a.all.Granges[[3]], HIF1a.all.Granges[[4]], HIF1a.all.Granges[[5]]) %>% 
  GenomicRanges::reduce(.) %>% #riduco gli overlap (semplifico)
  as.data.frame() %>% 
  dplyr::filter(nchar(as.vector(seqnames)) < 6) #rimuovo i cromosomi che non mi interessano
```

#salvo
```{r}
write_delim(HIF1a.summed, file = "HIF1a_merged_five_cell_lines.bed", delim = '\t', col_names = F)
```

#vediamo se stanno dentro alla signature di ipossia che avevo usato per l'RNAseq
```{r}
hypoxia_hallmarks_genes <-read.gmt("/home/alessio/Downloads/HALLMARK_HYPOXIA.v2023.2.Hs.gmt") %>% 
  dplyr::select(gene) %>% 
  unlist()

HIF1a.summed.genes <- unique(
  na.omit(
    HIF1a.summed.Anno@anno %>%
      as.data.frame() %>%
      dplyr::filter(str_detect(annotation, "Promoter")) %>% #switcho questo se voglio solo promotore
      dplyr::select(SYMBOL) %>%
      unlist()))

#salvo geni di HALLMARKS HYPOXIA condivisi con i picchi di HIF1a
write_lines(hypoxia_hallmarks_genes[hypoxia_hallmarks_genes %in% HIF1a.summed.genes], "Hallmarks_hypoxia_in_HIF1a_5_celllines_peaks_GTFv44.txt")
write_lines(hypoxia_hallmarks_genes[hypoxia_hallmarks_genes %in% HIF1a.summed.genes], "Hallmarks_hypoxia_in_HIF1a_5_celllines_peaks_GTFv44_only_promoter.txt")
```

##########################################################
#HIF1a e HIF2a putative binding regions generated by HOMER
##########################################################

```{r}
HIF1a_putative <- readPeakFile("HIF1a_all_putative_binding_sites.bed")
HIF2a_putative <- readPeakFile("HIF2a_all_putative_binding_sites.bed")
HIF1b_putative <- readPeakFile("HIF1b_all_putative_binding_sites.bed")

HIF1a_4_celllines <- readPeakFile("PRJNA823676/HIF1a.4.cell.lines.bed")
HIF2a_4_celllines <- readPeakFile("PRJNA823676/HIF2a.4.cell.lines.bed")
```

```{r}
png("HIF1a_HIF1b_Venn.png", res = 300, units = "cm", height = 10, width = 12)
makeVennDiagram(Peaks = list(HIF1a_putative, HIF1b_putative, HIF1a_4_celllines),
                NameOfPeaks = c("HIF1a motifs", "HIF1b motifs", "HIF1a 4 cell lines ChIP"),
                margin = 0.1,
                fill = c("darkred", "grey", "red"),
                alpha = 0.7)
dev.off()

png("HIF2a_HIF1b_Venn.png", res = 300, units = "cm", height = 10, width = 12)
makeVennDiagram(Peaks = list(HIF2a_putative, HIF1b_putative, HIF2a_4_celllines),
                NameOfPeaks = c("HIF2a motifs", "HIF1b motifs", "HIF2a 4 cell lines ChIP"),
                margin = 0.1,
                fill = c("darkblue", "grey", "blue"),
                alpha = 0.7)
dev.off()
```

```{r}
txdb = makeTxDbFromGFF("/media/alessio/Data/genome110/gencode/gencode.v44.annotation.gtf")

HIF1a_putative_anno <-  annotatePeak(HIF1a_putative, 
                                     TxDb = txdb,
                                     annoDb = "org.Hs.eg.db",
                                     tssRegion = c(-3000, 3000),
                                     flankDistance = 5000,
                                     level = "gene")

plotAnnoBar(HIF1a_putative_anno, title = "HIF1a_putative") + scale_fill_viridis_d()
plotDistToTSS(HIF1a_putative_anno)
```

